lib/workers/global/autodiscover.ts

```diff
@@ -30,7 +31,21 @@ export async function autodiscoverRepositories(
     return config;
   }
   if (config.autodiscoverFilter) {
-    discovered = discovered.filter(minimatch.filter(config.autodiscoverFilter));
+    if (isConfigRegex(config.autodiscoverFilter)) {
+      const autodiscoveryPred = configRegexPredicate(config.autodiscoverFilter);
+      if (!autodiscoveryPred) {
+        logger.error(
+          { autodiscoverFilter: config.autodiscoverFilter },
+          'Failed to parse regex pattern'
+        );
+        return config;
+      }
```

Comment:
Currently only presets and secrets are validated before repository discovery. https://github.com/renovatebot/renovate/blob/main/lib/workers/global/index.ts#L95-L100

Reply:
In that case, can we fatally error and abort?

Reply:
We can throw an error and abort this way, but it wouldn't make really a difference as outside of this function we simply loop over the found repositories and then fail as we have logged an error

This would the output look like if we throw an error instead of logging the error: 
```
DEBUG: Commits limit = null
DEBUG: Autodiscovering GitHub repositories
FATAL: Fatal error: Failed to parse regex pattern "/secustor/renovate**./"
       "err": {
         "message": "Failed to parse regex pattern \"/secustor/renovate**./\"",
         "stack": "Error: Failed to parse regex pattern \"/secustor/renovate**./\"\n    at autodiscoverRepositories (/home/secustor/repos/renovate/lib/workers/global/autodiscover.ts:37:15)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async Object.start (/home/secustor/repos/renovate/lib/workers/global/index.ts:103:14)\n    at async /home/secustor/repos/renovate/lib/renovate.ts:16:22"
       }
DEBUG: Renovate exiting
 INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 60,
           "logContext": "tuxtltgv5r0jwbaMUD3rp",
           "err": {
             "message": "Failed to parse regex pattern \"/secustor/renovate**./\"",
             "stack": "Error: Failed to parse regex pattern \"/secustor/renovate**./\"\n    at autodiscoverRepositories (/home/secustor/repos/renovate/lib/workers/global/autodiscover.ts:37:15)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async Object.start (/home/secustor/repos/renovate/lib/workers/global/index.ts:103:14)\n    at async /home/secustor/repos/renovate/lib/renovate.ts:16:22"
           },
           "msg": "Fatal error: Failed to parse regex pattern \"/secustor/renovate**./\""
         }
       ]

```

Reply:
Let's say that the bot is a member of 1000 repositories. And say that your intention is to define an autodiscover filter which limits it to 5 of them. Now let's say you make a typo in it. Wouldn't Renovate end up running over all 1000 before eventually exiting?

Reply:
As we [return immediately](https://github.com/renovatebot/renovate/pull/13243/files#diff-95ef0c8b78ce37c0f940c4328d6e34a3b67b5a940731efb9b5ddfa0df11d827dR41)  and not set `config.repositories` with the discovered repositories, this is not the case. Usually we do this at https://github.com/renovatebot/renovate/pull/13243/files#diff-95ef0c8b78ce37c0f940c4328d6e34a3b67b5a940731efb9b5ddfa0df11d827dR83 which is never reached in the error case.        

This test case is covered in https://github.com/renovatebot/renovate/pull/13243/files#diff-8152736b101d2b82ef650b8e341cd1cb87e8dafd4fe798df1a6c2fd01b97916cR101

In regard to the planned introduction of `autodiscoverFilter` as an array, this has to definitely be defined.